### PR TITLE
Replace "CSharp | VB | FSharp" usage with ".NET"

### PR DESCRIPTION
**Pull request by davkean**
This replaces all usage of specific languages in our components to the new ".NET" capability. This enables other .NET languages to opt into our feature set.

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/ProjectCapability.cs [hunk 1]
```diff
@@ -36,5 +36,6 @@ internal static class ProjectCapability
         public const string ProjectConfigurationsDeclaredDimensions = ProjectCapabilities.ProjectConfigurationsDeclaredDimensions;
         public const string LanguageService = ProjectCapabilities.LanguageService;
         public const string SortByDisplayOrder = ProjectCapabilities.SortByDisplayOrder;
+        public const string DotNet = ".NET";
```
Comment (Pilchie) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/ProjectCapability.cs]: Don't you need to remove the one on line 14?

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/ProjectCapability.cs]: yeah that would help.

**Comment by davkean** [comment]
Just testing this.

**Comment by davkean** [comment]
This breaks shared projects, which don't pull use our factory registration.

**Comment by davkean** [comment]
We need to make https://github.com/dotnet/project-system/issues/3311 before we check this in.

**Comment by davkean** [comment]
Ready for review, Shared Projects is fixing in: https://github.com/dotnet/project-system/pull/3521

**Pull request merged**